rasa/shared/core/domain.py

```diff
@@ -1916,6 +1917,25 @@ def _collect_actions_which_explicitly_need_domain(
 
         return action_names
 
+    @staticmethod
+    def _validate_response_ids(responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
+        """Validates that all response ids are unique.
+
+        Raises:
+            ValueError: if there are any non-unique response ids.
+        """
+        response_ids = set()
+        for _, response_variations in responses.items():
+            for response_variation in response_variations:
+                response_variation_id = response_variation.get("id")
+                if response_variation_id:
+                    if response_variation_id in response_ids:
+                        raise RasaException(
+                            f"Duplicate response id "
+                            f"'{response_variation_id}' defined in domain."
+                        )
+                    response_ids.add(response_variation_id)
```

Comment:
Shouldn't these response ids be returned (mapped to the right utter_name) and then passed to the constructor of Domain so they can be stored on the domain? Then you don't need to extract this again in `ActionBotResponse`? 🤔

Reply:
Well, I always need to filter IDs based on the condition slots. Pre-processing can only save time for responses which do not have condition set.

Reply:
If we do not need to do this I can just remove that code.